compiler/src/iree/compiler/Codegen/Common/LinalgOpInfo.cpp

```diff
@@ -57,51 +56,84 @@ static bool isTransposeMap(AffineMap map) {
   return false;
 }
 
+/// The default filter passes all op operands.
+static bool defaultTransposeMapFilter(AffineMap map) { return true; }
+
+LinalgOpInfo::LinalgOpInfo(linalg::LinalgOp linalgOp)
+    : transposeMapFilter(defaultTransposeMapFilter) {
+  computeInfo(linalgOp);
+}
+LinalgOpInfo::LinalgOpInfo(linalg::LinalgOp linalgOp,
+                           TransposeMapFilter transposeMapFilter)
```

Comment:
Great! (Nothing to address as part of this PR) Eventually this should move to a setter because otherwise this won't scale if we need more and more filters. Also, we shouldn't compute all the information at construction time.

Reply:
agreed.